packages/sds/src/message_channel/message_channel.ts

```diff
@@ -196,14 +334,22 @@ export class MessageChannel extends TypedEventEmitter<MessageChannelEvents> {
       this.timeReceived.set(message.messageId, Date.now());
     } else {
       this.deliverMessage(message);
+      this.safeDispatchEvent(MessageChannelEvent.MessageDelivered, {
+        detail: {
+          messageId: message.messageId,
+          sentOrReceived: "received"
+        }
+      });
     }
   }
 
   // https://rfc.vac.dev/vac/raw/sds/#periodic-incoming-buffer-sweep
+  // Note that even though this function has side effects, it is not async
+  // and does not need to be called through the queue.
   public sweepIncomingBuffer(): HistoryEntry[] {
```

Comment:
is this function intended to be used outside? 
I am not sure based on the comment 
and if there is a side-effect maybe renaming the function is better as from comment I still don't understand